com.unity.render-pipelines.universal/Editor/2D/ShaderGraph/Nodes/LightTextureNode.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using UnityEditor.Graphing;
+using UnityEditor.ShaderGraph;
+using UnityEditor.ShaderGraph.Drawing.Controls;
+using UnityEditor.ShaderGraph.Internal;
+using UnityEngine;
+
+namespace _2D.ShaderGraph
+{
+    enum BlendStyle
+    {
+        LightTex0,
+        LightTex1,
+        LightTex2,
+        LightTex3,
+    }
+
+    [Title("Input", "2D", "Light Texture")]
```

Comment:
Will this node will only work when you have a Sprite Custom Lit subtarget on the graph?

Reply:
it will work with any 2D sub targets... but the CustomLit adds a normal pass that's in the Lit target but not in the Unlit target...

Reply:
My worry here is that users will add this node to non-2D graphs, and be confused when it doesn't work.  We have the ability to flag nodes as only working on a specific Target, so it gives a nice warning tag message when used inappropriately.  It might be good to extend that to SubTargets as well, so we can flag this node as only working with the 2D subtargets...

Reply:
This check could be added by extending UniversalTarget.cs IsNodeAllowedByTarget() to also ask the subtarget if it allows the node.   The subtarget can check for a filter attribute similar to SRPFilterAttribute -- would have to add a new filter attribute like "SubTargetFilterAttribute",  or maybe more specifically for this use case:  "2DSubTargetOnlyAttribute"

Reply:
Done... please take a look